chapter-05/final/money/decimal.go

```diff
@@ -78,9 +99,9 @@ const (
 	ErrInputTooLarge = moneyError("input amount over 10^15 is too large")
 	// ErrOutputTooLarge is returned if the converted amount is too large, to protect against floating point errors.
 	ErrOutputTooLarge = moneyError("output amount is too large (over 10^15)")
+	// ErrInputTooPrecise is returned if the input amount is too precise, which could lead to precision issues.
+	ErrInputTooPrecise = moneyError("input amount should be at least 1 * precision")
```

Comment:
precision is something internal to the system. It shouldn't be exposed. "input includes subdivisions of the currency that are too small"